test/apps/iodemo/io_demo.cc

```diff
@@ -874,34 +942,27 @@ class DemoClient : public P2pDemoCommon {
     }
 
     void wait_for_responses(long max_outstanding) {
-        struct timeval tv_start = {};
-        bool timer_started      = false;
-        bool timer_finished     = false;
-        struct timeval tv_curr, tv_diff;
-        long count;
+        bool timer_started  = false;
+        bool timer_finished = false;
+        double start_time, curr_time, elapsed_time;
 
-        count = 0;
         while (((_num_sent - _num_completed) > max_outstanding) &&
                (_status == OK)) {
-            if ((count < 1000) || timer_finished) {
-                progress();
-                ++count;
+            progress();
```

Comment:
but then we will progress even up to 1000 more times if all reqs are completed?

Reply:
only if progress does some work, if ucp_worker returns 0, the loop is done.
IMO if we don't complete that work this round to keep numbers more accurate, it will affect next one any way.